src/connector/src/macros.rs

```diff
@@ -77,11 +77,13 @@ macro_rules! impl_split {
             }
 
             fn encode_to_bytes(&self) -> Bytes {
-                Bytes::from(ConnectorSplit::from(self).encode_to_vec())
+                let hex_string = hex::encode(ConnectorSplit::from(self).encode_to_vec());
+                Bytes::from(hex_string.into_bytes())
```

Comment:
This interface gives an intuition that any bytes are acceptable in the return value, but it does not according to the description in the PR body. So I think we should ensure this through type safety, instead of such fix. 🤔

There might be two choices:
- Use `bytea` for internal table schema, and no need to hex encode here.
- If still to keep the `varchar` type, make the return value to `String`, so that Rust ensures that it must be valid under UTF-8.

Reply:
Yes, after changing to `Bytea` it works.